code-studio/src/js/components/progress/course_progress.jsx

```diff
@@ -2,15 +2,25 @@
 
 var STAGE_TYPE = require('./types').STAGE_TYPE;
 var CourseProgressRow = require('./course_progress_row');
+var StageDetails = require('./stage_details');
 
 /**
  * Stage progress component used in level header and course overview.
  */
 var CourseProgress = React.createClass({
   propTypes: {
+    display: React.PropTypes.oneOf(['dots', 'list']).isRequired,
     stages: React.PropTypes.arrayOf(STAGE_TYPE)
   },
 
+  getRow: function (stage) {
+    if (this.props.display === 'dots') {
+      return <CourseProgressRow stage={stage} key={stage.name}/>;
+    } else {
+      return <StageDetails  stage={stage} key={stage.name}/>;
```

Comment:
We should update the [Style Guide](https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#react) if we always want a space before the closing `/>`.  Right now it says:
> We're okay with no space in self-closing tags, i.e. `<MyComponent/>` and `<MyComponent />` are both valid.